src/main/java/org/cryptomator/ui/mainwindow/MainWindowController.java

```diff
@@ -116,6 +129,34 @@ private void addVault(Path pathToVault) {
 		}
 	}
 
+	private boolean containsUnlockedVaultContent(Path path) {
+		return vaultListManager.list().stream().anyMatch(v -> path.startsWith(v.getAccessPoint()));
+	}
+
+	private Optional<Path> getCiphertextPath(Path path) {
+		return vaultListManager.list().stream() //
+				.filter(v -> path.startsWith(v.getAccessPoint())) //
+				.findAny() //
+				.map(v -> {
+					try {
+						var cleartextPathString = path.toString().substring(v.getAccessPoint().length());
+						if (!cleartextPathString.startsWith("/")) {
+							cleartextPathString = "/" + cleartextPathString;
+						}
+						var cleartextPath = Path.of(cleartextPathString);
+						return v.getCiphertextPath(cleartextPath);
+					} catch (IOException e) {
+						LOG.debug("Unable to get ciphertext path from path: {}", path);
+						return null;
+					}
+				});
+	}
+
+	private void revealPath(Path path) {
+		// Unable to use `application.get().getHostServices().showDocument()` here, because it does not reveal files, only folders.
```

Comment:
For a given path, how hard can it be to find out its parent? 😉

Reply:
But we want to_reveal a _file_ and not its parent. That's the whole point of this feature to find the encrypted file. If you just reveal the parent, you still won't know which encrypted file to select for restoring an old version.

Reply:
> But we want to reveal a _file_ and not its parent. That's the whole point of this feature to find the encrypted file. If you just reveal the parent, you still won't know which encrypted file to select for restoring an old version.

Finding the parent would help for selective sync, though, no? E.g. https://github.com/cryptomator/cryptomator/issues/1895